src/lib.rs

```diff
@@ -198,7 +246,33 @@ impl<'t> MachineDescriptor<'t> {
         timeout: Option<std::time::Duration>,
         port: u16,
     ) -> Result<Machine<'t>, Report> {
-        let mut sess = ssh::SessionBuilder::default();
+        #[cfg(feature = "openssh")]
+        let sess = self
+            .do_connect_ssh(username, key_path, timeout, port)
+            .await?;
+
+        Ok(Machine {
+            nickname: self.nickname,
+            public_ip: self.public_ip,
+            private_ip: self.private_ip,
+            public_dns: self.public_dns,
+            _tsunami: self._tsunami,
+            #[cfg(feature = "openssh")]
+            ssh: sess,
+            username: username.to_string(),
+            private_key: key_path.map(|path| path.to_path_buf()),
+        })
+    }
+
+    #[cfg(feature = "openssh")]
+    async fn do_connect_ssh(
```

Comment:
[nit] I understand the desire to minimize the surface area of the change, but if `connect_ssh` in some cases *doesn't* connect to SSH it's probably no longer a useful name.

Maybe `connect` or `initialize` and then you can have a private `connect_ssh` (what you call `do_connect_ssh` now)?